components/rootmail.yaml

```diff
@@ -370,54 +498,91 @@ Resources:
-                  }
-                ]),
-                "Type":"SearchableString",
-              },
-            }
-
-            ssm.create_ops_item(
-              Description=description,
-              OperationalData=operational_data,
-              Source=source,
-              Title=title,
-            )
+            log({
+              'event': event,
+              'level': 'debug',
+            })
+
+            for record in event['Records']:
+
+              id = record['ses']['mail']['messageId']
+              key = 'RootEmails/{key}'.format(key=id)
+              receipt = record['ses']['receipt']
+
+              log({
+                'id': id,
+                'level': 'debug',
+                'key': key,
+                'msg': 'processing mail',
+              })
+
+              verdicts = {
+                'dkim': receipt['dkimVerdict']['status'],
+                'spam': receipt['spamVerdict']['status'],
+                'spf': receipt['spfVerdict']['status'],
+                'virus': receipt['virusVerdict']['status'],
+              }
+
+              for k, v in verdicts.items():
+
+                if not v == 'PASS':
+
+                  log({
+                    'class': k,
+                    'id': id,
+                    'key': key,
+                    'level': 'warn',
+                    'msg': 'verdict failed - ops santa item skipped',
+                  })
+
+                  return
+
+              response = s3.get_object(
+                Bucket="${DomainBucket}",
+                Key=key,
+              )
+
+              msg = email.message_from_bytes(response["Body"].read(), policy=policy.default)
+
+              title=msg["subject"]
+              title=title[:1020] + " ..." * (len(title) > 1020)
+
+              description=msg.get_body("plain").get_content()
+              description=description[:1020] + " ..." * (len(description) > 1020)
+
+              source=recipient=event["Records"][0]["ses"]["mail"]["destination"][0]
+              source=source[:60] + ' ...' * (len(source) > 60)
+
+              operational_data={
+                "/aws/dedup":{
+                  "Value":json.dumps(
+                    {
+                      "dedupString":id,
+                    }
+                  ),
+                  "Type":"SearchableString",
+                },
+                "/aws/resources":{
+                  "Value":json.dumps([
+                    {
+                      "arn":"arn:aws:s3:::${DomainBucket.Arn}/{key}".format(key=key),
+                    }
+                  ]),
+                  "Type":"SearchableString",
+                },
+              }
+
+              ssm.create_ops_item(
+                Description=description,
+                OperationalData=operational_data,
+                Source=source,
+                Title=title,
+              )
+
+        def log(msg):
+          print(json.dumps(msg), flush=True)
```

Comment:
Python also has a native `logging` lib, which could be used. But this simple json dump is OK for me for now.